calico/openstack/mech_calico.py

```diff
@@ -100,26 +103,52 @@ def __init__(self):
         # initialize these properly when we first need them.
         self.db = None
         self.transport = None
-        self._state_initialized = False
+        self._my_pid = None
+        self._periodic_resync_greenlet = None
+        self._epoch = 0
 
-        # Start our resynchronization process.
-        eventlet.spawn(self.periodic_resync_thread)
+        # Make sure we initialise even if we don't see any API calls.
+        eventlet.spawn_after(STARTUP_DELAY_SECS, self._init_state)
 
     def _init_state(self):
         """
         Creates the connection state required for talking to the Neutron DB
         and to etcd. This is a no-op if it has been executed before.
         """
-        if self._state_initialized:
+        current_pid = os.getpid()
+        if self._my_pid == current_pid:
+            # We've initialised our PID and it hasn't changed since last time,
+            # nothing to do.
```

Comment:
It is a bit paranoid but I can't see a way it can go wrong, even if Neutron gets changed quite a bit. Even if our startup timer magically pops before the fork or Neutron completely changes its process model, we'll still catch it.

I only measure a 9ns penalty for the `getpid()` call vs field access; in the noise compared with DB access. 
```
$ python -m timeit -n 1000000 -s 'o = lambda x: x; o.foo=False' -- 'if o.foo: pass'
1000000 loops, best of 3: 0.0497 usec per loop
$ python -m timeit -n 1000000 -s 'from os import getpid' -- 'if getpid() == 1: pass'
1000000 loops, best of 3: 0.0506 usec per loop
```